### PR TITLE
Fixed the issue of `databricks labs ucx installations` and `databricks labs ucx manual-workspace-info`.

### LINKED ISSUE #785: Crash of `databricks labs ucx installations` command
Command crashes. 

```
$ databricks labs ucx installations
12:09:17  INFO [databricks.sdk] Using Databricks CLI authentication
12:09:17  INFO [d.labs.ucx] Fetching installations...
12:09:43  INFO [d.l.blueprint.parallel][detecting_installations_1] detecting installations 500/3332, rps: 38.547/sec
12:09:54  INFO [d.l.blueprint.parallel][detecting_installations_17] detecting installations 1000/3332, rps: 41.348/sec
12:10:08  INFO [d.l.blueprint.parallel][detecting_installations_8] detecting installations 1500/3332, rps: 39.046/sec
12:10:20  INFO [d.l.blueprint.parallel][detecting_installations_1] detecting installations 2000/3332, rps: 39.809/sec

...

12:10:55  INFO [d.l.blueprint.parallel][detecting_installations_5] detecting installations 3332/3332, rps: 39.414/sec
12:10:55 ERROR [d.l.blueprint.parallel] More than half 'detecting installations' tasks failed: 2% results available (59/3332). Took 0:01:24.563580
12:10:55 ERROR [d.l.ucx.installations] ManyError: Detected 1 failures: ScannerError: mapping values are not allowed here
  in "<byte string>", line 7, column 45:
     ... : Serverless Shared Endpoint (ID: 475b94ddc7cd52110...SDZACVZX
                                         ^
Error: unexpected end of JSON input
```

Possibly a duplicate of https://github.com/databrickslabs/ucx/issues/784

### PR DESCRIPTION
**Pull request by HariGS-DB**
## Changes
Fixing the issue of `databricks labs ucx installations` and `databricks labs ucx manual-workspace-info`.
When the workspace config file is scanned, if the config file is not following yaml format (ex containing additional ":"), it throws an error which is not caught. This change catches the YAML error and raises IllegalState msg for for_user function in Confi

### Linked issues

Resolves #785 #784 

### Functionality 

- Modified installer.InstallationManager.for_user function to add YAMLError catch and raise a custom msg as a IllegalState Exception

### Tests

- manually tested
- installed the ucx from a branch using databricks labs install ucx@bug/incorrect_yaml_format
- added an entry in the config file for warehouse id which contains ":"
- 
image

- ran command databricks labs ucx installations
- output:
- Skipped that entry as expected and no failure
image
- ran command databricks labs ucx manual-workspace-info
- output:
- skipped the entry as expected and no failure
image

### CODE REVIEW

**Comment by nfx** [comment]
@HariGS-DB run it on e2-demo-field-eng

File: tests/unit/installer/test_installation_manager.py [hunk 1]
```diff
@@ -63,6 +63,15 @@ def test_corrupt_config(mocker):
     assert len(user_installations) == 0
 
 
+def test_corrupt_config_yaml(mocker):
+    ws = mocker.patch("databricks.sdk.WorkspaceClient.__init__")
```
Comment (nfx) [inline tests/unit/installer/test_installation_manager.py:67]: Create autospec

File: tests/unit/installer/test_installation_manager.py [hunk 2]
```diff
@@ -63,6 +63,15 @@ def test_corrupt_config(mocker):
...
+    ws.users.list.return_value = [User(user_name="foo")]
+    ws.workspace.download.return_value = io.StringIO("version: 2\ntacl: extra colon: test")
+    installation_manager = InstallationManager(ws)
```
Comment (nfx) [inline tests/unit/installer/test_installation_manager.py:70]: Do the test with corrupt yaml

**Review by nfx — CHANGES_REQUESTED** [review]
+ check on e2-demo-field-eng

**Comment by HariGS-DB** [comment]
output of databricks labs ucx installations from e2 field demo no error
image
output of databricks labs ucx manual-workspace-info no error
image

File: tests/unit/installer/test_installation_manager.py (hunk 2 above)
Comment (HariGS-DB) [reply tests/unit/installer/test_installation_manager.py:83]: yes i have created a string with corrupt yaml having extra ":" re-wrote the test cases to showing raises right Exception

**Review by nfx — APPROVED** [review]
👍

**Pull request merged**